kubectl-plugin/test/e2e/kubectl_ray_job_submit_test.go

```diff
@@ -31,77 +37,129 @@ var _ = Describe("Calling ray plugin `job submit` command on Ray Job", func() {
-		output, err = cmd.CombinedOutput()
-		Expect(err).ToNot(HaveOccurred())
+		assertRayJobSucceeded(ctx, namespace, rayJobName, cmdOutputJobID)
+	})
 
-		Expect(string(output)).To(Equal("SUCCEEDED"))
+	It("succeed in submitting RayJob with headNodeSelectors and workerNodeSelectors", func() {
+		// To avoid port conflict, kill any existing port-forward process on 8265
+		KillPortForwardOn8265()
+		runtimeEnvFilePath := path.Join(kubectlRayJobWorkingDir, runtimeEnvSampleFileName)
 
-		// Retrieve Job Deployment Status
-		cmd = exec.Command("kubectl", "get", "--namespace", namespace, "rayjob", "rayjob-sample", "-o", "jsonpath={.status.jobDeploymentStatus}")
-		output, err = cmd.CombinedOutput()
-		Expect(err).ToNot(HaveOccurred())
+		ctx, cancel := context.WithTimeout(context.Background(), 3*time.Minute)
+		defer cancel()
+
+		cmd := exec.CommandContext(
+			ctx,
+			"kubectl", "ray", "job", "submit",
+			"--namespace", namespace,
+			"--name", rayJobName,
+			"--runtime-env", runtimeEnvFilePath,
+			"--head-cpu", "1",
+			"--head-memory", "2Gi",
+			"--worker-cpu", "1",
+			"--worker-memory", "2Gi",
+			"--head-node-selectors", "kubernetes.io/os=linux",
+			"--worker-node-selectors", "kubernetes.io/os=linux",
+			"--",
+			"python",
+			entrypointSampleFileName,
+		)
+		output, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred())
+		// Retrieve the Job ID from the output
+		cmdOutputJobID := extractRayJobID(string(output))
 
-		Expect(string(output)).To(Equal("Complete"))
+		rayJob := assertRayJobSucceeded(ctx, namespace, rayJobName, cmdOutputJobID)
+		// Retrieve Job Head Node Selectors
+		Expect(rayJob.Spec.RayClusterSpec.HeadGroupSpec.Template.Spec.NodeSelector["kubernetes.io/os"]).To(Equal("linux"))
+		// Retrieve Job Worker Node Selectors
+		Expect(rayJob.Spec.RayClusterSpec.WorkerGroupSpecs[0].Template.Spec.NodeSelector["kubernetes.io/os"]).To(Equal("linux"))
 	})
 })
+
+// extractRayJobID extracts the Ray Job ID from the kubectl ray job submit output.
+//
+// Use regex to extract the job ID from the output.
+// The output is expected to be like:
+//
+//	Current status: RUNNING (RayJob: rayjob-sample, JobID: raysubmit_XBBxrNHztQa1rHpL)
+//	Current status: SUCCEEDED (RayJob: rayjob-sample, JobID: raysubmit_XBBxrNHztQa1rHpL)
+//	Job raysubmit_XBBxrNHztQa1rHpL finished with status SUCCEEDED.
+//
+// It returns the first matched Job ID string like "raysubmit_XBBxrNHztQa1rHpL".
+func extractRayJobID(output string) string {
+	// Match any string that starts with "raysubmit_" followed by alphanumeric characters.
+	regexExp := regexp.MustCompile(`raysubmit_[a-zA-Z0-9]+`)
+
+	// Find the first match in the output
+	matches := regexExp.FindStringSubmatch(output)
+
+	// Return the matched Job ID if found, otherwise return an empty string
+	if len(matches) >= 1 {
+		return matches[0]
+	}
+	return ""
+}
+
+func assertRayJobSucceeded(ctx context.Context, namespace, jobName, cmdOutputJobID string) (rayjob rayv1.RayJob) {
```

Comment:
we should not limit the function to specific status only.

```suggestion
func getAndCheckRayJob(ctx context.Context, namespace, name, expectedJobID, expectedJobStatus, expectedJobDeploymentStatus string) (rayjob rayv1.RayJob) {
```